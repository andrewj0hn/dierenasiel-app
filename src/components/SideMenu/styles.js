// @flow
import { StyleSheet } from 'react-native';
import {
  TOP_LAYER, SIDE_MENU_BACKGROUND_COLOR, SIDE_MENU_ITEM_TEXT, BORDER_GREY, REGULAR,
} from '../../constants/Styles';

export default StyleSheet.create({
  container: {
    flexDirection: 'column',
    position: 'absolute',
    zIndex: TOP_LAYER,
    top: 0,
    right: 0,
    bottom: 0,
    justifyContent: 'center',
    backgroundColor: SIDE_MENU_BACKGROUND_COLOR,
    borderLeftWidth: 0.5,
    borderLeftColor: BORDER_GREY,
  },
  label: {
    color: SIDE_MENU_ITEM_TEXT,
    fontSize: 16,
    fontWeight: REGULAR,
    letterSpacing: 0.4,
    marginVertical: 10,
    paddingVertical: 10,
    textAlign: 'center',
  },
});
