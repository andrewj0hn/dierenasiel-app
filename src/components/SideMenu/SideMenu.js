// @flow
import React from 'react';
import { View, Text, TouchableHighlight } from 'react-native';
import styles from './styles';
import MENU_ITEMS from '../../constants/MenuItems';
import { SIDE_MENU_ITEM_HIGHLIGHTED, SIDE_MENU_WIDTH, WHITE } from '../../constants/Styles';

type Props = {
  sideMenuIsOpen: boolean,
  onSideMenuItemPress: Function,
  routeName: string,
};

const defaultProp = {
  menuIsOpen: false,
};

const SideMenu = ({ sideMenuIsOpen, onSideMenuItemPress, routeName }: Props) => (
  <View style={[styles.container, { width: sideMenuIsOpen ? SIDE_MENU_WIDTH : 0 }]}>
    { MENU_ITEMS.map(({ label, screen }) => (
      <TouchableHighlight
        key={label}
        underlayColor={SIDE_MENU_ITEM_HIGHLIGHTED}
        onPress={() => onSideMenuItemPress(screen)}
        style={{ backgroundColor: screen === routeName ? SIDE_MENU_ITEM_HIGHLIGHTED : WHITE }}
      >
        <Text style={styles.label}>{label}</Text>
      </TouchableHighlight>
    ))}
  </View>
);

SideMenu.defaultProp = defaultProp;

export default SideMenu;
