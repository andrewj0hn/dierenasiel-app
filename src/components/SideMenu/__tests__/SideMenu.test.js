import React from 'react';
import renderer from 'react-test-renderer';
import SideMenu from '../SideMenu';

test('Menu renders correctly', () => {
  const tree = renderer.create(<SideMenu />).toJSON();
  expect(tree).toMatchSnapshot();
});
