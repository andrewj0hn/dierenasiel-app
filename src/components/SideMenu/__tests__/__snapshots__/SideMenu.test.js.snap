// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFF",
        "borderLeftColor": "#DFDFDF",
        "borderLeftWidth": 0.5,
        "bottom": 0,
        "flexDirection": "column",
        "justifyContent": "center",
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 1,
      },
      Object {
        "width": 0,
      },
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#FFF",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#3A3A3A",
          "fontSize": 16,
          "fontWeight": "400",
          "letterSpacing": 0.4,
          "marginVertical": 10,
          "paddingVertical": 10,
          "textAlign": "center",
        }
      }
    >
      Home
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#FFF",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#3A3A3A",
          "fontSize": 16,
          "fontWeight": "400",
          "letterSpacing": 0.4,
          "marginVertical": 10,
          "paddingVertical": 10,
          "textAlign": "center",
        }
      }
    >
      Honden
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#FFF",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#3A3A3A",
          "fontSize": 16,
          "fontWeight": "400",
          "letterSpacing": 0.4,
          "marginVertical": 10,
          "paddingVertical": 10,
          "textAlign": "center",
        }
      }
    >
      Katten
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#FFF",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#3A3A3A",
          "fontSize": 16,
          "fontWeight": "400",
          "letterSpacing": 0.4,
          "marginVertical": 10,
          "paddingVertical": 10,
          "textAlign": "center",
        }
      }
    >
      Pension
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#FFF",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#3A3A3A",
          "fontSize": 16,
          "fontWeight": "400",
          "letterSpacing": 0.4,
          "marginVertical": 10,
          "paddingVertical": 10,
          "textAlign": "center",
        }
      }
    >
      Contact
    </Text>
  </View>
</View>
`;
